packages/phpstan-static-type-mapper/src/TypeMapper/ArrayTypeMapper.php

```diff
@@ -122,24 +121,7 @@ private function isGenericArrayCandidate(ArrayType $arrayType): bool
         if ($arrayType->getKeyType() instanceof NeverType) {
             return false;
         }
-
-        // make sure the integer key type is not natural/implicit array int keys
-        $keysArrayType = $arrayType->getKeysArray();
-        if (! $keysArrayType instanceof ConstantArrayType) {
-            return true;
-        }
-
-        foreach ($keysArrayType->getValueTypes() as $key => $keyType) {
-            if (! $keyType instanceof ConstantIntegerType) {
-                return true;
-            }
-
-            if ($key !== $keyType->getValue()) {
-                return true;
-            }
-        }
```

Comment:
Removing this now causes to add `int` key type, when not explicitly used

```php
/**
 * @var array<int, string>  
 */
private const VALUE = [
   'something'
];
```

:x: 

<br>

It should be:

```php
/**
 * @var string[]
 */
private const VALUE = [
   'something'
];
```

:heavy_check_mark: